ambry-server/src/test/java/com/github/ambry/server/AmbryServerSecurityServiceTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.server;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.commons.ServerMetrics;
+import com.github.ambry.config.ServerConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.protocol.AmbryRequests;
+import com.github.ambry.rest.MockChannelHandlerContext;
+import com.github.ambry.rest.MockRestRequest;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.rest.RestServiceErrorCode;
+import com.github.ambry.rest.RestServiceException;
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.ThrowingConsumer;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.embedded.EmbeddedChannel;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class AmbryServerSecurityServiceTest {
+  private final ServerSecurityService serverSecurityService =
+      new AmbryServerSecurityService(new ServerConfig(new VerifiableProperties(new Properties())),
+          new ServerMetrics(new MetricRegistry(), AmbryRequests.class, AmbryServer.class));
+
+  @Test
+  public void validateConnectionTest() throws Exception {
+    //ctx is null
+    TestUtils.assertException(IllegalArgumentException.class,
+        () -> serverSecurityService.validateConnection(null).get(), null);
+
+    //success case
+    EmbeddedChannel channel = new EmbeddedChannel();
+    ChannelHandlerContext ctx = new MockChannelHandlerContext(channel);
+    try {
+      serverSecurityService.validateConnection(ctx, (r, e) -> {
+        Assert.assertNull("result not null", r);
+        Assert.assertNull("exception not null", e);
+      });
+    } catch (Exception e) {
+      Assert.fail("unexpected exception happened" + e);
```

Comment:
I know some ambry test cases may have had this assert.fail pattern, but I think it is generally better to let the exception be thrown in test cases instead of catching it and wrapping it in an assert. Throwing the exception gives us the full stack trace (and still fails the test) which can be helpful in tracing down the issue if the test happens to fail.

Reply:
Thanks for the suggestion. removed the try/catch.